src/test/java/software/amazon/encryption/s3/S3AsyncEncryptionClientTest.java

```diff
@@ -829,6 +835,111 @@ public void testAsyncInstructionFileConfig() {
+                        .enableInstructionFilePutObject(true)
+                        .build())
+                .kmsKeyId(KMS_KEY_ID)
+                .enableMultipartPutObject(true)
+                .credentialsProvider(credentials)
+                .build();
+        PutObjectRequest request = PutObjectRequest.builder()
+                .bucket(BUCKET)
+                .key(objectKey)
+                .build();
+        CompletableFuture<PutObjectResponse> putObjectResponse = v3Client.putObject(request, AsyncRequestBody.fromString(input));
+        putObjectResponse.join();
+
+        assertNotNull(putObjectResponse);
+
+        ResponseBytes<GetObjectResponse> instructionFileResponse = wrappedClient.getObjectAsBytes(builder -> builder
+                .bucket(BUCKET)
+                .key(objectKey + ".instruction")
+                .build());
+        assertNotNull(instructionFileResponse);
+        assertTrue(instructionFileResponse.response().metadata().containsKey("x-amz-crypto-instr-file"));
+
+        CompletableFuture<ResponseBytes<GetObjectResponse>> futureGetObj = v3Client.getObject(builder -> builder
+                .bucket(BUCKET)
+                .key(objectKey)
+                .build(), AsyncResponseTransformer.toBytes());
+        ResponseBytes<GetObjectResponse> getResponse = futureGetObj.join();
+        assertNotNull(getResponse);
+        assertEquals(input, getResponse.asUtf8String());
+
+        deleteObject(BUCKET, objectKey, v3Client);
+
+        v3Client.close();
+    }
+    @Test
+    public void testAsyncInstructionFileConfigMultipartWithOptions() {
+        final String objectKey = appendTestSuffix("test-multipart-async-instruction-file-config-options");
+        final String input = "SimpleTestOfV3EncryptionClient";
+
+        AwsCredentialsProvider credentials = DefaultCredentialsProvider.create();
+        S3Client wrappedClient = S3Client.create();
+        S3AsyncClient v3Client = S3AsyncEncryptionClient.builder()
+                .instructionFileConfig(InstructionFileConfig.builder()
+                        .instructionFileClient(wrappedClient)
+                        .enableInstructionFilePutObject(true)
+                        .build())
+                .kmsKeyId(KMS_KEY_ID)
+                .enableMultipartPutObject(true)
+                .credentialsProvider(credentials)
+                .build();
+        CreateMultipartUploadRequest multipartUploadRequest = CreateMultipartUploadRequest.builder()
+                .bucket(BUCKET)
+                .key(objectKey)
+                .storageClass(StorageClass.STANDARD_IA)
+                .build();
+        assertNotNull(multipartUploadRequest);
+
+        PutObjectRequest putObjectRequest = ConvertSDKRequests.convertRequest(multipartUploadRequest);
+
+        assertNotNull(putObjectRequest);
+        assertEquals(putObjectRequest.storageClassAsString(), multipartUploadRequest.storageClassAsString());
```

Comment:
We don't need this here, this is already tested by the unit tests in `ConvertSDKRequests`